### PR TITLE
VZ-7683: increase ut coverage for verrazzano reconcile

### PR DESCRIPTION
**Pull request by karan56625**
Test cases added for reconcile package.
Now package coverage is around ~84

### CODE REVIEW

File: platform-operator/controllers/verrazzano/reconcile/status.go [hunk 1]
```diff
@@ -263,11 +263,11 @@ func (r *Reconciler) checkComponentReadyState(vzctx vzcontext.VerrazzanoContext)
 	for _, comp := range registry.GetComponents() {
 		spiCtx, err := spi.NewContext(vzctx.Log, r.Client, vzctx.ActualCR, nil, r.DryRun)
 		if err != nil {
-			spiCtx.Log().Errorf("Failed to create component context: %v", err)
+			vzctx.Log.Errorf("Failed to create component context: %v", err)
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/reconcile/status.go:266]: Why did this have to change?

File: platform-operator/controllers/verrazzano/reconcile/status.go [hunk 2]
```diff
@@ -263,11 +263,11 @@ func (r *Reconciler) checkComponentReadyState(vzctx vzcontext.VerrazzanoContext)
...
 			return false, err
 		}
 		if comp.IsEnabled(spiCtx.EffectiveCR()) && cr.Status.Components[comp.Name()].State != installv1alpha1.CompStateReady {
-			spiCtx.Log().Progressf("Waiting for component %s to be ready", comp.Name())
+			vzctx.Log.Progressf("Waiting for component %s to be ready", comp.Name())
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/reconcile/status.go:270]: Same question as above

File: platform-operator/controllers/verrazzano/reconcile/uninstall_test.go [hunk 3]
```diff
@@ -6,9 +6,16 @@ package reconcile
...
 	"fmt"
+	"reflect"
 	"testing"
 	"time"
 
+	"github.com/golang/mock/gomock"
+	"github.com/verrazzano/verrazzano/pkg/helm"
+	vzos "github.com/verrazzano/verrazzano/pkg/os"
+	"github.com/verrazzano/verrazzano/platform-operator/mocks"
+	controllerruntime "sigs.k8s.io/controller-runtime"
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/reconcile/uninstall_test.go:17]: Minor comment but reorganize imports like this
```
<list-of-internal-golang-packages>

<list-of-verrazzano-internal-imports>

<all-other-thridparty-imports>
```
Upto you if you want to do this in this pass but it's a good practice to follow.

File: platform-operator/mocks/runtime_controller_mock.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (adiforluls) [inline platform-operator/mocks/runtime_controller_mock.go:1]: Why was this file needed? We already have a controller_mock.go inside mocks package, can you move/generate whatever you added to that file if its missing in there and then use that file?

Comment (adiforluls) [inline platform-operator/mocks/runtime_controller_mock.go:1]: Why was this file needed? We already have a controller_mock.go inside mocks package, can you move/generate whatever you added to that file if its missing in there and then use that file?

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 1 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/reconcile/status.go:266]: If there is any error from spi.NewContext, spiCtx will be nil and spiCtx.Log() will panic.

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 2 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/reconcile/status.go:270]: As we changed the line 263, better to use same log variable for  here too.

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 1 above)
Comment (adiforluls) [reply platform-operator/controllers/verrazzano/reconcile/status.go:266]: Ok

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 2 above)
Comment (adiforluls) [reply platform-operator/controllers/verrazzano/reconcile/status.go:270]: Probably should restore to spiCtx log here

File: platform-operator/mocks/runtime_controller_mock.go (hunk 4 above)
Comment (karan56625) [reply platform-operator/mocks/runtime_controller_mock.go:1]: I tried to generate mocks from two different files in one output file. But using reflection mode of mockgen, I was to not able to pass multiple files in one statement. So, need to have two statements for mockgen.
 //go:generate mockgen -destination=../../../mocks/controller_mock.go -package=mocks -copyright_file=../../../hack/boilerplate.go.txt sigs.k8s.io/controller-runtime/pkg/client Client,StatusWriter
//go:generate mockgen -destination=../../../mocks/runtime_controller_mock.go -package=mocks -copyright_file=../../../hack/boilerplate.go.txt sigs.k8s.io/controller-runtime/pkg/controller Controller

Is there any way?

File: platform-operator/controllers/verrazzano/reconcile/uninstall_test.go (hunk 3 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/reconcile/uninstall_test.go:17]: yeah, can do in this pass.

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 2 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/reconcile/status.go:270]: ok

File: platform-operator/mocks/runtime_controller_mock.go (hunk 4 above)
Comment (adiforluls) [reply platform-operator/mocks/runtime_controller_mock.go:1]: My question is more about why did you need this new file at all? We should have all the things needed to mock a controller in controller_mock.go. If not, then anything new should go in there.

Comment (karan56625) [reply platform-operator/mocks/runtime_controller_mock.go:1]: I need to mock Controller interface from `package sigs.k8s.io/controller-runtime/pkg/controller` for test function `TestReconcilerInitForVzResource`.  `controller_mock.go ` does't contain Controller mock. It has only mock for package `controller-runtime/pkg/client` not for `package sigs.k8s.io/controller-runtime/pkg/controller`

Comment (adiforluls) [reply platform-operator/mocks/runtime_controller_mock.go:1]: Oh, yes. Probably add a make target here then https://github.com/verrazzano/verrazzano/blob/4b5a4c66c7d77677f3028d9c665b3256a21f74c5/platform-operator/Makefile#L106
If you add a make target for this file and then run it, it should auto-generate stuff for you. The mock files are all auto-generated afaik.

Comment (karan56625) [reply platform-operator/mocks/runtime_controller_mock.go:1]: yeah, they are auto-generated. I thought go generate will take care of that as we added comment `//go:generate`. Will update in make file too.

File: platform-operator/controllers/verrazzano/reconcile/uninstall_test.go (hunk 3 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/reconcile/uninstall_test.go]: done

File: platform-operator/controllers/verrazzano/reconcile/status.go (hunk 2 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/reconcile/status.go]: done

File: platform-operator/mocks/runtime_controller_mock.go (hunk 4 above)
Comment (karan56625) [reply platform-operator/mocks/runtime_controller_mock.go:1]: done! I had to change `boilerplate.go.txt`. As additional "//"  was getting added to generated mock files.

**Review by adiforluls — APPROVED** [review]
